### ISSUE TITLE
theme: missing drop shadow when using blurred background

### ISSUE THREAD

**Issue by tecandrew** [zed-industries/zed#15383]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

latest version of Catppuccin theme exposed a strange behavior when blurred background is set. the **drop shadow** is missing on MacOS when this is enabled. seems like it happens across the board amongst any theme that uses this?

tested using this in `settings.json`

```json
"experimental.theme_overrides": {
  "background.appearance": "blurred"
}
```

ref: https://github.com/catppuccin/zed/issues/27
may be related to https://github.com/zed-industries/zed/issues/10993 ?

### Environment

```
Zed: v0.145.1 (Zed)
OS: macOS 14.5.0
Memory: 16 GiB
Architecture: aarch64
```

### If applicable, add mockups / screenshots to help explain present your vision of the feature

(on plain white background)

<img width="1218" alt="image" src="https://github.com/user-attachments/assets/d753c2c6-f75d-4af0-85b6-8f14748c929b">

<img width="1250" alt="image" src="https://github.com/user-attachments/assets/d735abea-ec08-4efc-8755-dff6103d4102">

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by jansol** [comment]
Yes, as mentioned in #10993 window shadows are explicitly disabled for transparent windows in GPUI since they completely destroyed resizing and exposé performance - blur is based on a private API used by Terminal.app, which *does* have drop shadows with seemingly no performance problems. But since the APIs are private, there is no documentation for them and I have no idea what Terminal.app is doing differently (all my information is essentially from what `winit` does, which in turn is reverse engineered from somewhere else). If you find a way to have both transparency/blur and shadows, please do open a PR to fix it.

**Comment by hfjallemark** [comment]
I can still reproduce this issue.
